src/main/java/htmlpublisher/HtmlPublisher.java

```diff
@@ -76,10 +77,15 @@ private static void writeFile(ArrayList<String> lines, File path) throws IOExcep
 
     public ArrayList<String> readFile(String filePath) throws java.io.FileNotFoundException,
             java.io.IOException {
+        return readFile(filePath, this.getClass());
+    }
+    
+    public static ArrayList<String> readFile(String filePath, Class<?> publisherClass) 
+            throws java.io.FileNotFoundException, java.io.IOException {
         ArrayList<String> aList = new ArrayList<String>();
 
         try {
-            final InputStream is = this.getClass().getResourceAsStream(filePath);
+            final InputStream is = publisherClass.getResourceAsStream(filePath);
```

Comment:
Overkill. Just hardcode `HtmlPublisher.class`. (The original code would have triggered a FindBugs violation, BTW; it is not safe for subclassing from another package.)